### ISSUE TITLE
fortran_c_interop bug in transfer statement 

### ISSUE THREAD

**Issue by jedwards4b** [CrayLabs/SmartRedis#166]
# Description
The transfer statement at line 74 of fortran_c_interop.F90 produces an error if length(character_array_f(i)) == 0 when using the intel 19.1.1 compiler.   The error is: When the SOURCE argument to the intrinsic function TRANSFER has size greater than zero, the MOLD argument must not be an array with elements of size zero.

# How to reproduce
steps to reproduce the bug  

# Expected behavior

Here is a proposed solution:
```
@@ -71,11 +71,13 @@ subroutine convert_char_array_to_c(character_array_f, character_array_c, string_
 

   ! Copy the character into a c_char and create pointers to each of the strings
   do i=1,n_strings
-    character_array_c(i) = transfer(character_array_f(i),character_array_c(i))
-    string_ptrs(i) = c_loc(character_array_c(i))
+     if(string_lengths(i) > 0) then
+        character_array_c(i) = transfer(character_array_f(i),character_array_c(i))
+        string_ptrs(i) = c_loc(character_array_c(i))
+     endif


```
# System
 - OS: Linux
 - OS version:
 - Commit/version of library: v0.2.0

**Comment by ashao** [comment]
I agree with the proposed solution.

**Comment by billschereriii** [comment]
I agree with the solution and with the tweak

**Comment by billschereriii** [comment]
Addressed in PR #170

**Comment by billschereriii** [comment]
@jedwards4b I couldn't list you as a formal reviewer, but if you could look at the PR, I'd appreciate your feedback.
